### DISCUSSION TITLE
[Bug]: Short description of the issue

### DISCUSSION THREAD

**Discussion by R1YMOLIM** [JaKooLit/NixOS-Hyprland#24] [Q&A]
![Screenshot_07-Dec_23-59-08_13324](https://github.com/user-attachments/assets/b39b4076-8059-4f6a-9dea-881e0368fff4)
Hello guys i'm configuring conf.nix and get this error help me pls!

┌─[r1y@NixOS] - [~] - [Sat Dec 07, 23:50]
└─[$] <> sudo nixos-rebuild -I nixos-config=/home/r1y/NixOS-Hyprland/hosts/NixOS/config.nix switch             

building Nix...
building the system configuration...
error:
       … while calling the 'head' builtin
         at /nix/store/nr5nl3zwzl02x3rnikjbry3s5xy7bm1d-source/lib/attrsets.nix:1574:11:
         1573|         || pred here (elemAt values 1) (head values) then
         1574|           head values
             |           ^
         1575|         else

       … while evaluating the attribute 'value'
         at /nix/store/nr5nl3zwzl02x3rnikjbry3s5xy7bm1d-source/lib/modules.nix:846:9:
          845|     in warnDeprecation opt //
          846|       { value = addErrorContext "while evaluating the option `${showOption loc}':" value;
             |         ^
          847|         inherit (res.defsFinal') highestPrio;

       … while evaluating the option `system.build.toplevel':

       … while evaluating definitions from `/nix/store/nr5nl3zwzl02x3rnikjbry3s5xy7bm1d-source/nixos/modules/system/activation/top-level.nix':

       … while evaluating the option `assertions':

       … while evaluating definitions from `/nix/store/nr5nl3zwzl02x3rnikjbry3s5xy7bm1d-source/nixos/modules/system/boot/systemd.nix':

       … while evaluating the option `systemd.services':

       … while evaluating definitions from `/nix/store/nr5nl3zwzl02x3rnikjbry3s5xy7bm1d-source/nixos/modules/services/logging/logrotate.nix':

       … while evaluating the option `services.logrotate.enable':

       … while evaluating the option `users.groups':

       … while evaluating definitions from `/nix/store/nr5nl3zwzl02x3rnikjbry3s5xy7bm1d-source/nixos/modules/config/resolvconf.nix':

       … while evaluating the option `networking.resolvconf.enable':

       … while evaluating the option `environment.etc':

       … while evaluating definitions from `/nix/store/nr5nl3zwzl02x3rnikjbry3s5xy7bm1d-source/nixos/modules/services/networking/ssh/sshd.nix':

       … while evaluating the option `users.users':

       … while evaluating definitions from `/home/r1y/NixOS-Hyprland/hosts/NixOS/users.nix':

       … while evaluating the module argument `username' in "/home/r1y/NixOS-Hyprland/hosts/NixOS/users.nix":

       (stack trace truncated; use '--show-trace' to show the full, detailed trace)

       error: attribute 'username' missing
       at /nix/store/nr5nl3zwzl02x3rnikjbry3s5xy7bm1d-source/lib/modules.nix:515:28:
          514|         addErrorContext (context name)
          515|           (args.${name} or config._module.args.${name})
             |                            ^
          516|       ) (functionArgs f);
┌─[r1y@NixOS] - [~] - [Sat Dec 07, 23:58]
└─[$] <>

**Comment by JaKooLit** [comment]
moved into discussion as this is not a project issue
